examples/vision/sam.py

```diff
@@ -0,0 +1,601 @@
+"""
+Title: Segment Anything Model with 🤗Transformers
+Author: [Sayak Paul](https://twitter.com/RisingSayak) & [Merve Noyan](https://twitter.com/mervenoyann)
+Date created: 2023/07/11
+Last modified: 2023/07/11
+Description: Implementing Segment Anything Model using 🤗Transformers.
+Accelerator: GPU
+"""
+"""
+## Introduction
+
+Large language models (LLMs) make it easy for the end users to apply them to various
+applications through "prompting". For example if we wanted an LLM to predict the
+sentiment of the following sentence -- "That movie was amazing, I thoroughly enjoyed it"
+-- we'd do prompt the LLM with something like:
+
+> What's the sentiment of the following sentence: "That movie was amazing, I thoroughly
+enjoyed it"?
+
+In return, the LLM would return sentiment token.
+
+But when it comes to visual recognition tasks, how can we engineer "visual" cues to
+prompt foundation vision models? For example, we could have an input image and prompt the
+model with bounding box on that image and ask it to perform segmentation. The bounding
+box would serve as our visual prompt here.
+
+In the [Segment Anything Model](https://segment-anything.com/) (dubbed as SAM),
+researchers from Meta extended the space of language prompting to visual prompting. SAM
+is capable of performing zero-shot segmentation with a prompt input, inspired by large
+language models. The prompt here can be a set of foreground/background points, free text,
+a box or a mask. There are many downstream segmentation tasks, including semantic
+segmentation and edge detection. The goal of SAM is to enable all of these downstream
+segmentation tasks through prompting.
+
+In this notebook, we'll learn how to use the SAM model from 🤗 Transformers for performing
+inference and fine-tuning.
+
+
+"""
+
+"""
+## Installation
+"""
+
+"""shell
+!pip install -q git+https://github.com/huggingface/transformers
+"""
+
+"""
+## SAM in a few words
+
+SAM has the following components:
+
+|
+![](https://scontent.fccu31-1.fna.fbcdn.net/v/t39.2365-6/338558258_1349701259095991_4358060436604292355_n.png?_nc_cat=104&ccb=1-7&_nc_sid=ad8a9d&_nc_ohc=f0B0G6TtZG0AX-2gBP1&_nc_ht=scontent.fccu31-1.fna&oh=00_AfDSPCChiBRx_ZmE_krAlqsfSIZU3BUmRJcZIc4LVZa93w&oe=64737F89)
+![](https://scontent.fccu31-1.fna.fbcdn.net/v/t39.2365-6/338558258_1349701259095991_4358060436604292355_n.png?_nc_cat=104&ccb=1-7&_nc_sid=ad8a9d&_nc_ohc=f0B0G6TtZG0AX-2gBP1&_nc_ht=scontent.fccu31-1.fna&oh=00_AfDSPCChiBRx_ZmE_krAlqsfSIZU3BUmRJcZIc4LVZa93w&oe=64737F89)
```

Comment:
Seems like this one is extra

Reply:
Also, this URL tends to behave weirdly. Let's use: https://i.imgur.com/oLfdwuB.png.